### PR TITLE
Add python linter job

### PR DESCRIPTION
**Pull request by dosas**
and fix warnings

@bastian-src I hope the added .pylintrc does not cause problems in foreman or ruby packaging?

### CODE REVIEW

File: .github/workflows/python.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+name: Pylint
+# yamllint disable-line rule:truthy
+on: [push, pull_request]
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.8", "3.9", "3.11", "3.12"]
```
Comment (bastian-src) [inline .github/workflows/python.yaml:11]: The current Salt versions 3006 and 3007 support Python 3.10.x only, so I'd drop 3.8 and 3.9 here (and add 3.10):

```suggestion
        python-version: ["3.10", "3.11", "3.12"]
```

source: https://docs.saltproject.io/salt/install-guide/en/latest/topics/salt-python-version-support.html

**Review by bastian-src — COMMENTED** [review]
Thanks for this PR @dosas!!
I think its a great idea to lint our custom Python Salt modules.
Could you please split your changes into two commits (adding GHA, adding Python fixes + pylintrc)?

@evgeni do you have comments too?

File: salt/report_upload/srv/salt/_runners/foreman_report_upload.py [hunk 2]
```diff
@@ -3,6 +3,7 @@
 Uploads reports from the Salt job cache to Foreman
 '''
 from __future__ import absolute_import, print_function, unicode_literals
+from builtins import str
```
Comment (evgeni) [inline salt/report_upload/srv/salt/_runners/foreman_report_upload.py:6]: Why would one need to import str?

File: .github/workflows/python.yaml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+name: Pylint
+# yamllint disable-line rule:truthy
+on: [push, pull_request]
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.10", "3.11", "3.12"]
```
Comment (evgeni) [inline .github/workflows/python.yaml:11]: Should this include 3.6 and 3.9 for el8 and el9 clients?

Comment (dosas) [reply .github/workflows/python.yaml:11]: I will put here whatever people want I have no strong opinion on that. See previous comment from @bastian-src https://github.com/theforeman/smart_proxy_salt/pull/100#discussion_r1869519277

File: salt/report_upload/srv/salt/_runners/foreman_report_upload.py (hunk 2 above)
Comment (dosas) [reply salt/report_upload/srv/salt/_runners/foreman_report_upload.py:6]: To get rid of the `unicode = str` line https://github.com/theforeman/smart_proxy_salt/blob/master/salt/report_upload/srv/salt/_runners/foreman_report_upload.py#L30-L31 that the linter has an issue with.

It is a python 2/3 compatibility fix: https://python-future.org/compatible_idioms.html#unicode

Comment (dosas) [reply salt/report_upload/srv/salt/_runners/foreman_report_upload.py:6]: I do not know that but there is still a lot of compatibility fixes in the code. 

The PR is not about removing python 2 compatibility but we can do that in a separate commit.

File: .github/workflows/python.yaml (hunk 3 above)
Comment (evgeni) [reply .github/workflows/python.yaml:11]: Ah, interesting, so even on those old platforms it wants py310

File: salt/report_upload/srv/salt/_runners/foreman_report_upload.py (hunk 2 above)
Comment (evgeni) [reply salt/report_upload/srv/salt/_runners/foreman_report_upload.py:6]: Ok

Comment (dosas) [reply salt/report_upload/srv/salt/_runners/foreman_report_upload.py:6]: `pyupgrade --py36-plus --keep-percent-format `find . -name "*.py" -type f`` should do the job :D

**Comment by bastian-src** [comment]
I assume there are not further comments :) thanks @dosas for the PR and @evgeni for commenting!

**Pull request merged**